src/flint/fmpz_mat.jl

```diff
@@ -1810,81 +1781,91 @@ end
 #
 ###############################################################################
 
-function zero!(z::ZZMatrix)
+function zero!(z::ZZMatrixOrPtr)
   ccall((:fmpz_mat_zero, libflint), Nothing,
         (Ref{ZZMatrix},), z)
   return z
 end
 
-function one!(z::ZZMatrix)
+function one!(z::ZZMatrixOrPtr)
   ccall((:fmpz_mat_one, libflint), Nothing,
         (Ref{ZZMatrix},), z)
   return z
 end
 
-function neg!(z::ZZMatrix, w::ZZMatrix)
+function neg!(z::ZZMatrixOrPtr, w::ZZMatrixOrPtr)
   ccall((:fmpz_mat_neg, libflint), Nothing, (Ref{ZZMatrix}, Ref{ZZMatrix}), z, w)
   return z
 end
 
-function add!(z::ZZMatrix, x::ZZMatrix, y::ZZMatrix)
+function add!(z::ZZMatrixOrPtr, x::ZZMatrixOrPtr, y::ZZMatrixOrPtr)
   ccall((:fmpz_mat_add, libflint), Nothing,
         (Ref{ZZMatrix}, Ref{ZZMatrix}, Ref{ZZMatrix}), z, x, y)
   return z
 end
 
-function sub!(z::ZZMatrix, x::ZZMatrix, y::ZZMatrix)
+function sub!(z::ZZMatrixOrPtr, x::ZZMatrixOrPtr, y::ZZMatrixOrPtr)
   ccall((:fmpz_mat_sub, libflint), Nothing,
         (Ref{ZZMatrix}, Ref{ZZMatrix}, Ref{ZZMatrix}), z, x, y)
   return z
 end
 
-function mul!(z::ZZMatrix, x::ZZMatrix, y::ZZMatrix)
+function mul!(z::ZZMatrixOrPtr, x::ZZMatrixOrPtr, y::ZZMatrixOrPtr)
   ccall((:fmpz_mat_mul, libflint), Nothing,
         (Ref{ZZMatrix}, Ref{ZZMatrix}, Ref{ZZMatrix}), z, x, y)
   return z
 end
 
-function mul!(y::ZZMatrix, x::Int)
-  mul!(y, y, x)
-end
-
-function mul!(z::ZZMatrix, y::ZZMatrix, x::Int)
+function mul!(z::ZZMatrixOrPtr, a::ZZMatrixOrPtr, b::Int)
   ccall((:fmpz_mat_scalar_mul_si, libflint), Nothing,
-        (Ref{ZZMatrix}, Ref{ZZMatrix}, Int), z, y, x)
+        (Ref{ZZMatrix}, Ref{ZZMatrix}, Int), z, a, b)
   return z
 end
 
-function mul!(y::ZZMatrix, x::ZZRingElem)
-  mul!(y, y, x)
+function mul!(z::ZZMatrixOrPtr, a::ZZMatrixOrPtr, b::UInt)
+  ccall((:fmpz_mat_scalar_mul_ui, libflint), Nothing,
+        (Ref{ZZMatrix}, Ref{ZZMatrix}, UInt), z, a, b)
+  return z
 end
 
-function mul!(z::ZZMatrix, y::ZZMatrix, x::ZZRingElem)
+function mul!(z::ZZMatrixOrPtr, a::ZZMatrixOrPtr, b::ZZRingElemOrPtr)
   ccall((:fmpz_mat_scalar_mul_fmpz, libflint), Nothing,
-        (Ref{ZZMatrix}, Ref{ZZMatrix}, Ref{ZZRingElem}), z, y, x)
+        (Ref{ZZMatrix}, Ref{ZZMatrix}, Ref{ZZRingElem}), z, a, b)
   return z
 end
 
-function addmul!(z::ZZMatrix, y::ZZMatrix, x::ZZRingElem)
+mul!(z::ZZMatrixOrPtr, a::ZZMatrixOrPtr, b::Integer) = mul!(z, a, flintify(b))
+mul!(z::ZZMatrixOrPtr, a::Union{Integer, ZZRingElemOrPtr}, b::ZZMatrixOrPtr) = mul!(z, b, a)
```

Comment:
```suggestion
mul!(z::ZZMatrixOrPtr, a::Union{IntegerUnionOrPtr}, b::ZZMatrixOrPtr) = mul!(z, b, a)
```

Reply:
You probably meant this?
```suggestion
mul!(z::ZZMatrixOrPtr, a::IntegerUnionOrPtr, b::ZZMatrixOrPtr) = mul!(z, b, a)
```

Reply:
oops, yeah of course